File: src/main/java/com/procurement/point/model/entity/ReleaseEntity.java
package com.procurement.point.model.entity;

import java.time.LocalDateTime;
import java.time.ZoneOffset;
import java.util.Date;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.cassandra.core.cql.Ordering;
import org.springframework.data.cassandra.core.cql.PrimaryKeyType;
import org.springframework.data.cassandra.core.mapping.Column;
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn;
import org.springframework.data.cassandra.core.mapping.Table;

@Getter
@Setter
@Table("notice_release")
public class ReleaseEntity {

    @PrimaryKeyColumn(name = "cp_id", type = PrimaryKeyType.PARTITIONED)
    private String cpId;

    @PrimaryKeyColumn(name = "oc_id", type = PrimaryKeyType.CLUSTERED)
    private String ocId;

    @PrimaryKeyColumn(name = "release_date", type = PrimaryKeyType.CLUSTERED, ordering = Ordering.DESCENDING)
    private Date releaseDate;

    @PrimaryKeyColumn(name = "release_id", type = PrimaryKeyType.CLUSTERED)
    private String releaseId;

    @Column(value = "stage")
    private String stage;

    @Column(value = "json_data")
    private String jsonData;

    public LocalDateTime getReleaseDate() {
        return LocalDateTime.ofInstant(releaseDate.toInstant(), ZoneOffset.UTC);
    }
}

File: src/main/kotlin/com/procurement/point/model/entity/ReleaseEntity.kt
package com.procurement.point.model.entity

import org.springframework.data.cassandra.core.cql.Ordering
import org.springframework.data.cassandra.core.cql.PrimaryKeyType
import org.springframework.data.cassandra.core.mapping.Column
import org.springframework.data.cassandra.core.mapping.PrimaryKeyColumn
import org.springframework.data.cassandra.core.mapping.Table
import java.util.*

@Table(value = "notice_release")
data class ReleaseEntity(

        @PrimaryKeyColumn(name = "cp_id", type = PrimaryKeyType.PARTITIONED)
        val cpId: String,

        @PrimaryKeyColumn(name = "oc_id", type = PrimaryKeyType.CLUSTERED)
        val ocId: String,

        @PrimaryKeyColumn(name = "release_date", type = PrimaryKeyType.CLUSTERED, ordering = Ordering.DESCENDING)
        val releaseDate: Date,

        @PrimaryKeyColumn(name = "release_id", type = PrimaryKeyType.CLUSTERED)
        val releaseId: String,

        @Column(value = "stage")
        val stage: String,

        @Column(value = "json_data")
        val jsonData: String
)
